### ISSUE TITLE
Use official golang client for tasks

### ISSUE THREAD

**Issue by t-lin** [Multi-Tier-Cloud/docker-driver#4]
- [x] PullImage()
- [x] StopContainer()
- [x] DeleteContainer()
- [x] RestartContainer()
- [x] ResizeContainer()
- [x] RunContainer() when setting CPU/Mem

**Comment by kebahpng** [comment]
Commit 329b574 fixes this issue.

PullImage() which uses an iostream doesn't work without reading the stream content. Right now not saving the response from ioutil.ReadAll() is the workaround, so may need a better solution.

**Comment by t-lin** [comment]
@andrew-cc-chen I was able to reproduce and debug the issue you reported. Looks like it's a timing issue.

If you insert a sleep for 1 second (without calling `ReadAll()`), you'll see the test image (busybox, 1.22 MB) is properly downloaded. If you then repeat the test with a larger image (I used tlin/iamv3-mariadb, which is almost 464 MB), then sleeping for 1 second isn't enough, and the issue you reported re-appears.

Basically, without the `ReadAll()`, the function ends before the image is fully transferred. It seems like Docker's `ImagePull()` function is asynchronous, and so we should wait until an EOF is read from the reader object before we exit the function. Luckily, the `ReadAll()` function has this built-in (i.e. it returns only after EOF is read), so we should definitely call that each time.
